zebra-consensus/src/block/check.rs

```diff
@@ -25,3 +30,40 @@ pub fn is_coinbase_first(block: &Block) -> Result<(), Error> {
     }
     Ok(())
 }
+
+/// [3.9]: https://zips.z.cash/protocol/protocol.pdf#subsidyconcepts
+pub fn is_subsidy_correct(block: &Block) -> Result<(), Error> {
+    let height = block.coinbase_height().unwrap();
```

Comment:
What situations can a block not have a `coinbase_height()`? We might need this to be a recoverable error if we think this could happen normally.

```suggestion
    let height = block.coinbase_height().ok_or("block has no coinbase height");
```

if we think this indicates a bug however lets document that

```suggestion
    let height = block.coinbase_height().expect("always called on blocks with a coinbase height");
```

Reply:
The genesis block does not have a coinbase height. All other blocks do, as a consensus rule.

Reply:
By https://github.com/ZcashFoundation/zebra/blob/main/zebra-consensus/src/block.rs#L89 i think it is safe to expect the blocks evaluated to always with a coinbase height. Added `expect()` version at https://github.com/ZcashFoundation/zebra/pull/1051/commits/cfcbd70eb3cb41c7809b7fe7ed8b9b0f7eb21f92